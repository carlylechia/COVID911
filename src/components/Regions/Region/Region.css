.states-container {
  display: flex;
  justify-content: space-between;
}

.states-container:nth-child(even) {
  background: rgb(222, 49, 99);
  padding: 1rem 2rem;
  color: aliceblue;
  font-size: larger;
}

.states-container:nth-child(odd) {
  background: #d70040;
  padding: 1rem 2rem;
  color: azure;
  font-size: larger;
}

#loading {
  margin: 2rem auto;
}
