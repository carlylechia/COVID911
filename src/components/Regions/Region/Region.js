import PropTypes from 'prop-types';
import './Region.css';

const Region = ({ name, newCases }) => (
  <div className="f4 fw4 states-container pa5-ns pa3 pt3 pb3">
    <p className="f4 fw4 white">{name}</p>
    <p className="f4 fw4 tr pr3 white">{newCases}</p>
  </div>
);
Region.propTypes = {
  name: PropTypes.string.isRequired,
  newCases: PropTypes.number.isRequired,
};
export default Region;
