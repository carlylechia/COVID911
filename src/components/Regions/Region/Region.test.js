import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../Redux/createStore';
import Region from './Region';

describe('Region component test', () => {
  test('Region snapshot', () => {
    const props = {
      name: 'region 1',
      newCases: 50,
    };
    const {
      name, newCases,
    } = props;
    const region = render(
      <Router>
        <Provider store={store}>
          <Region
            name={name}
            newCases={newCases}
          />
        </Provider>
      </Router>,
    );
    expect(region).toMatchSnapshot();
  });
});
