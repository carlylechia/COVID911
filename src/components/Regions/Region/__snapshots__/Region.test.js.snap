// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Region component test Region snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="f4 fw4 states-container pa5-ns pa3 pt3 pb3"
      >
        <p
          class="f4 fw4 white"
        >
          region 1
        </p>
        <p
          class="f4 fw4 tr pr3 white"
        >
          50
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="f4 fw4 states-container pa5-ns pa3 pt3 pb3"
    >
      <p
        class="f4 fw4 white"
      >
        region 1
      </p>
      <p
        class="f4 fw4 tr pr3 white"
      >
        50
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
